src/components/core/Markdown/MarkdownShortcuts.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import Bold from './Bold';
+import Link from './Link';
+import List from './List';
+
+export default function MarkdownShortcuts({
+  textAreaRef,
+  onChange, // The onChange from the TextArea component. This ensures that we track the same event handlers (e.g. a collector's note change) via the addition of just markdown
+}: {
+  // textAreaRef:
+  //   | ((instance: HTMLTextAreaElement) => void)
+  //   | React.MutableRefObject<HTMLTextAreaElement | null>;
+  textAreaRef: React.MutableRefObject<HTMLTextAreaElement | null>;
+  onChange: React.ChangeEventHandler<HTMLTextAreaElement>;
+}) {
+  // Trigger onChange whenever textAreaRef.current changes
+  useEffect(() => {
+    if (textAreaRef.current) {
+      const event = new CustomEvent('change', {
+        detail: { value: textAreaRef.current.value },
+      });
+      Object.defineProperty(event, 'target', {
+        value: textAreaRef.current,
+      });
+      // textAreaRef.current.dispatchEvent(event);
+      // onChange(event as React.ChangeEvent<HTMLTextAreaElement>);
+      onChange(event);
+    }
+  }, [textAreaRef?.current?.value, onChange, textAreaRef]);
```

Comment:
so i read your PR description but i'm a bit confused as to why this is necessary 🤔 

if i understand this correctly, it's saying "if the text area's value changes, fire the onChange event".

is there a reason we can't do this from the parent?

e.g.
```js
// TextArea.tsx

    const textAreaRef = ref || _ref;

    const handleChange = useCallback((event) => {
      const event = new CustomEvent(...)
      onChange(event)
    }, [onChange])

    return (
      <>
        <StyledTextArea
          onChange={handleChange}
```

i'm probably missing something!

Reply:
re: typescript, i think you just have to cast it the way you did with `as`

Reply:
ahhh i think i get it. basically the markdown causes changes to the textArea ref... but that isn't necessarily picked up by the parent, so the parent wouldn't necessarily call `onChange`. is that right?

Reply:
Yep, exactly! Here's an example using the code you sent above:

https://user-images.githubusercontent.com/13339581/168489104-afe00170-ef0c-4c55-940b-8e7faf65c03a.mov

So this is the best way to handle it that I can think of

Reply:
@Robinnnnn when I pass `event as React.ChangeEvent<HTMLTextAreaElement>` I get the following:

```
Conversion of type 'CustomEvent<{ value: string; }>' to type 'ChangeEvent<HTMLTextAreaElement>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'CustomEvent<{ value: string; }>' is missing the following properties from type 'ChangeEvent<HTMLTextAreaElement>': nativeEvent, isDefaultPrevented, isPropagationStopped, persistts(2352)
```

We can fix this by instead passing `event as unknown as React.ChangeEvent<HTMLTextAreaElement>`... Feeling sort of hacky but I'm stuck here :/

Reply:
good news - took a look at this today and think I found a way to get it to work without the useEffect here!

it appears like the underlying issue is using `textArea.value = newText` to manually update the value. This doesn't trigger the event and onChange handler, as you noted.

I got textArea to detect that its value changed by using the native value setter method instead, then dispatching the input event manually, like so:

![Screen Shot 2022-05-16 at 20 03 04](https://user-images.githubusercontent.com/80802871/168579167-8f58a9b2-578f-4aa2-bcb5-5e2d999168d6.png)

[Reference 1](https://stackoverflow.com/a/61110332) [Reference 2](https://stackoverflow.com/a/46012210)

This triggered the TextArea's onChange handler without the useEffect as desired.

I also put this logic in a helper function called `setValueAndTriggerOnChange`. This way we can do:
`setValueAndTriggerOnChange(textArea, newText)
instead of
`textArea.value = newText;`
whenever we manually set the text area's value (in Bold.tsx, List.tsx, etc) and it automatically gets detected by the onChange handler.

lmk what you think! I think this moves the fix closer to the source of the issue, and leaves us with 1 less effect to think about.

Bonus - I noticed that with the original useEffect implementation, it worked correctly for Markdown Shortcut clicks, but the onChange handler was triggered twice for normal keypresses - I believe once by the TextArea's onChange and once by the useEffect, so this eliminates that too 

I also tweaked the stackoverflow example to remove `window` from `window.HTMLTextAreaElement.prototype` since we're trying to avoid adding usages of `window` for SSR purposes - looks like it still works but might be worth testing a bit more.

I opened a PR with these changes against this branch, #753 , feel free to refer to it or just merge it! I only applied it to `Bold.tsx` so you'll have to add /adapt it to List and Links.